rules/do-you-know-the-common-design-patterns-part-1/rule.md

```diff
@@ -12,40 +12,42 @@ authors:
   url: https://ssw.com.au/people/adam-stephensen
 - title: Damian Brady
   url: https://ssw.com.au/people/damian-brady
+- title: Dhruv Mathur
+  url: https://ssw.com.au/people/dhruv-mathur
 related: []
 redirects:
 - do-you-know-the-common-design-patterns-(part-1)
 
 ---
 
-Design patterns are useful for ensuring [common design principles](/do-you-know-the-common-design-principles-part-1)are being followed.  They help make your code consistent, predictable, and easy to maintain.
+Design patterns are useful for ensuring [common design principles](/do-you-know-the-common-design-principles-part-1) are being followed.  They help make your code consistent, predictable, and easy to maintain.
 
 <!--endintro-->
 
 There are a very large number of Design Patterns, but here are a few important ones.
 
-- **IOC** | [Inversion of Control](http&#58;//en.wikipedia.org/wiki/Inversion_of_control)  
-Control of the object coupling is the responsibility of the caller, not the class.
+* **IOC** | [Inversion of Control](http&#58;//en.wikipedia.org/wiki/Inversion_of_control)  
+In this pattern, control over the instantiation and management of objects is inverted, meaning that these responsibilities are handed over to an external framework like a DI container instead of being managed by the classes themselves. This separation enhances flexibility and decouples all the classes in the system.
 
-- **DI** | [Dependency Injection](http&#58;//en.wikipedia.org/wiki/Dependency_injection)  
-Dependencies are "injected" into the dependent object rather than the object depending on concretions.
+* **DI** | [Dependency Injection](http&#58;//en.wikipedia.org/wiki/Dependency_injection)  
+DI is a form of IoC where dependencies are provided to objects rather than created by them, one instance of the dependency can be used by many. This pattern also reduces dependency coupling between components since the instantiation is handled enternally, making the system easier to manage and test.
```

Comment:
Do you mean externally or internally?

```suggestion
DI is a form of IoC where dependencies are provided to objects rather than created by them, one instance of the dependency can be used by many. This pattern also reduces dependency coupling between components since the instantiation is handled externally, making the system easier to manage and test.
```